agent_build/package_builder/builder_inside_docker/Dockerfile

```diff
@@ -0,0 +1,18 @@
+ARG BASE_IMAGE_NAME
+#FROM frozen_binary_and_fpm_builder
+FROM $BASE_IMAGE_NAME
+
+# Add all files that are involved in the package build process.
+RUN mkdir /scalyr-agent-2
+ADD ./scalyr_agent /scalyr-agent-2/scalyr_agent
+ADD ./agent_build /scalyr-agent-2/agent_build
+ADD ./config /scalyr-agent-2/config
+ADD ./docker /scalyr-agent-2/docker
+ADD ./monitors /scalyr-agent-2/monitors
+ADD ./VERSION /scalyr-agent-2/VERSION
+
+ADD ./tests/distribution_tests/test_scalyr-agent.py /scalyr-agent-2/tests/distribution_tests/test_scalyr-agent.py
```

Comment:
There is some overlap between this test and AMI tests, would it make sense to also run all the checks we run for AMI tests inside this Docker environment as well?

And going forward, I assume we still plan to keep AMI tests, right?

Reply:
Yes, but now we can move everything from ami tests into python script, make a frozen binary from it and send that frozen binary to EC2 instances, so we won't have to maintain different sell scripts.

Reply:
Yeah, Python code should hopefully be easier to maintain, test, etc.

But we still need to ensure we cover the same path as AMI (this means also covering the installer script + various AMI images).

Reply:
Yes, for now, the plan is to keep everything in AMI tests as it is but just to remove the deployment shell scripts and replace them with python "frozen" scripts, with the same logic in it.